src/coreclr/jit/fginline.cpp

```diff
@@ -1433,6 +1433,13 @@ void Compiler::fgInsertInlineeBlocks(InlineInfo* pInlineInfo)
             // Save the basic block flags from the retExpr basic block.
             iciCall->gtInlineCandidateInfo->retExpr->AsRetExpr()->bbFlags = pInlineInfo->retBB->bbFlags;
         }
+
+        if (bottomBlock != nullptr)
+        {
+            // We've split the iciblock into two and the RET_EXPR was possibly moved to the bottomBlock
+            // so let's update its flags with retBB's ones
+            bottomBlock->bbFlags |= pInlineInfo->retBB->bbFlags & BBF_COMPACT_UPD;
+        }
```

Comment:
According to the comment above, the basic block flags update should've happened later in `fgUpdateInlineReturnExpressionPlaceHolder` when replacing `GT_RET_EXPR`. Why did it not happen?

Reply:
@erozenfeld sorry for the delay, didn't notice your comment.
So from my understanding we were supposed to save BBF flags here https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/fginline.cpp#L1434

but we lose connection between CALL and RET_EXPR somehow, I feel like it happens because RET_EXPR is re-used. So initially we inserted RET_EXPR in `Test` for `Calle1` but after `Calle1` was inlined we kept it to then re-use for `Callee2`
```csharp
    [MethodImpl(MethodImplOptions.NoInlining)]
    private static void Test(int[] x)
    {
        try
        {
             Callee1(x);
        }
        catch
        {
        }
    }

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static int Callee1(int[] x) => Callee2(x, 0);

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static int Callee2(int[] x, int index)
    {
        if (x == null)
            Callee3();

        return x.Length;
    }
```
while my fix works I guess I better investigate why we didn't update `iciCall->gtInlineCandidateInfo->retExpr` (currently it's some garbage value)

Reply:
Yeah, it's a tricky area and I had to touch it several times. I think it's worth spending a bit of time to understand why we didn't update `iciCall->gtInlineCandidateInfo->retExpr`.